crates/sui-types/src/transaction.rs

```diff
@@ -868,11 +873,13 @@ fn write_sep<T: Display>(
     items: impl IntoIterator<Item = T>,
     sep: &str,
 ) -> std::fmt::Result {
-    let mut xs = items.into_iter().peekable();
-    while let Some(x) = xs.next() {
-        if xs.peek().is_some() {
+    let mut start = true;
+    let xs = items.into_iter().peekable();
+    for x in xs {
+        if !start {
```

Comment:
Whoops, messed this up didn't I. 
But I think the intent was to be 
```suggestion
    let mut xs = items.into_iter().peekable();
    while let Some(x) = xs.next() {
        write!(f, "{x}")?; // MOVED HERE
        if xs.peek().is_some() {
            write!(f, "{sep}")?;
        }
    }
    Ok(())
```
So it doesn't have a trailing or leading separator